pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java

```diff
@@ -245,6 +240,28 @@ private void checkCompatible(SchemaAndMetadata existingSchema, SchemaData newSch
         });
     }
 
+    @Override
+    public CompletableFuture<Void> checkConsumerCompatibility(String schemaId, SchemaData schemaData,
+                                                              SchemaCompatibilityStrategy strategy) {
+        return getSchema(schemaId).thenCompose(existingSchema -> {
+            if (existingSchema != null && !existingSchema.schema.isDeleted()) {
+                    if (strategy == SchemaCompatibilityStrategy.BACKWARD ||
+                            strategy == SchemaCompatibilityStrategy.FORWARD ||
+                            strategy == SchemaCompatibilityStrategy.FORWARD_TRANSITIVE ||
+                            strategy == SchemaCompatibilityStrategy.FULL) {
+                        return checkCompatibilityWithLatest(schemaId, schemaData, SchemaCompatibilityStrategy.BACKWARD);
+                    } else {
+                        return checkCompatibilityWithAll(schemaId, schemaData, strategy);
+                    }
+            } else {
+                return FutureUtil.failedFuture(
+                        new IncompatibleSchemaException(
+                                "Trying to subscribe with incompatible schema"
+                        ));
```

Comment:
Shall we need to check the schema is exist, i think this can be handle in  checkCompatibilityWithLatest and checkCompatibilityWithAll ?